src/aiovodafone/api.py

```diff
@@ -414,23 +414,39 @@ async def _reset(self) -> bool:
 
         return False
 
-    async def _login_json(self, username: str, password: str) -> bool:
+    async def _login_json(
+        self, username: str, password: str, useChallenge: bool = False
+    ) -> bool:
         """Login via json page"""
 
-        payload = {
-            "LoginName": username,
-            "LoginPWD": password,
-        }
+        if useChallenge:
+            return_dict = await self._get_sercomm_page("/data/login.json")
+            challenge = return_dict["challenge"]
+            _LOGGER.debug("challenge: <%s>", challenge)
+            payload = {
+                "LoginName": username,
+                "LoginPWD": hashlib.sha256(
+                    bytes(password + challenge, "utf-8")
+                ).hexdigest(),
+                "challenge": challenge,
+            }
+        else:
+            payload = {
+                "LoginName": username,
+                "LoginPWD": password,
+            }
+
         reply_json = await self._post_sercomm_page("/data/login.json", payload)
-        _LOGGER.debug("Login result: %s[%s]", LOGIN[int(str(reply_json))], reply_json)
+        reply_int = int(str(reply_json))
+        _LOGGER.debug("Login result: %s[%s]", LOGIN[reply_int], reply_json)
 
-        if reply_json == "1":
+        if reply_int == 1:
```

Comment:
This won't work with Vodafone WiFi 6 Station (RHG3006)

Reply:
Original code [use](https://github.com/chemelli74/aiovodafone/blob/85041a1c26fe917a2c0a843c12e9502ed26b4d60/src/aiovodafone/api.py#L426) LOGIN[int(str(reply_json))], I just make reply_int = int(str(reply_json)) from it and use.